Tests/SonarQube.MSBuild.Tasks.IntegrationTests/TargetsTests/RoslynTargetsTests.cs

```diff
@@ -92,37 +68,38 @@ public void Roslyn_Settings_ValidSetup_OverrideExistingSettings_AbsolutePath()
             BuildAssertions.AssertTargetSucceeded(result, TargetConstants.OverrideRoslynAnalysisTarget);
 
             // Check the error log and ruleset properties are set
-            string sourceRulesetFilePath = GetDummyRulesetFilePath();
-            this.TestContext.AddResultFile(sourceRulesetFilePath);
-
-            string targetDir = result.ProjectStateAfterBuild.GetPropertyValue(TargetProperties.TargetDir);
-            string expectedErrorLog = Path.Combine(targetDir, ErrorLogFileName);
+            AssertErrorLogIsSetBySonarQubeTargets(result);
+            AssertExpectedResolvedRuleset(result, "d:\\my.ruleset");
+            AssertExpectedItemValuesExists(result, TargetProperties.AdditionalFilesItemType, expectedAdditionalFiles);
+            AssertExpectedItemValuesExists(result, TargetProperties.AnalyzerItemType, expectedAssemblies);
 
-            AssertExpectedAnalysisProperties(result, expectedErrorLog, sourceRulesetFilePath, GetDummySonarLintXmlFilePath());
             BuildAssertions.AssertWarningsAreNotTreatedAsErrorsNorIgnored(result);
-
-            // Check the analyzer properties are set as expected
-            List<string> expectedAnalyzers = new List<string>(GetSonarLintAnalyzerFilePaths());
-            AssertExpectedItemValuesExists(result, TargetProperties.AnalyzerItemType, expectedAnalyzers.ToArray());
         }
 
         [TestMethod]
-        [Description("Checks the ruleset and analyzers list correctly override existing settings")]
-        public void Roslyn_Settings_ValidSetup_OverrideExistingSettings_RelativePath()
+        [Description("Checks that a config file with no analyzer settings does not cause an issue")]
+        public void Roslyn_Settings_SettingsMissing_NoError()
```

Comment:
New test - config file with no analyzer settings should mean the build has no analyzers or additional files set.